### PR TITLE
Game speeding up for every 10 points is implemented

### PR DESCRIPTION
**Pull request by Hermix-art**
Additional logic was added to Waiter class, providing possibility to change game speed after game cycle. When game score achieved is multiple of 10, the delay decreases by 100 ms. (if current delay is higher o equals 100 ms), what causes growing level of game difficulty.

Implementation is based on timeout decrease for .sleep(long timeout) method of current thread.

### CODE REVIEW

**Comment by Winetq** [comment]
Hi @Hermix-art 
As far as I remember static fields should be all in upper case (in the WaiterTest class). Please correct me if I'm wrong :).

File: src/test/java/com/epam/prejap/tetris/game/WaiterTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    public void negativeDelayShouldNotThrowException() {
+        Waiter waiter = new Waiter(-100);
+        waiter.waitForIt();
+    }
+
+    @DataProvider
+    private static Object[][] constantDelays() {
+        return new Object[][]{
+                {1, delay},
```
Comment (TJB4epam) [inline src/test/java/com/epam/prejap/tetris/game/WaiterTest.java:72]: If the second param is always "delay" why we put it in the DPs? Why not have a constant and use it in the tests as needed

**Review by TJB4epam — CHANGES_REQUESTED** [review]
Love the factories. Hate the lack of groups.

**Comment by TJB4epam** [comment]
Has conflicts. Also, look carefully if someone didn't add `testng.xml` file.

**Comment by TJB4epam** [comment]
See previous comment.

File: src/main/java/com/epam/prejap/tetris/game/Waiter.java [hunk 2]
```diff
@@ -2,18 +2,52 @@
...
 import java.util.concurrent.TimeUnit;
 
+/**
+ * Causes delay between game iterations and speeds up the game
+ *
+ * @author Herman Kulik
+ */
 public class Waiter {
 
-    private final int milliseconds;
```
[the remark this answers is not in the document]
Comment (Hermix-art) [reply src/main/java/com/epam/prejap/tetris/game/Waiter.java:7]: I left it as final, but added an additional variable, which represents delay adjustment

File: src/test/java/com/epam/prejap/tetris/game/WaiterTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import org.testng.annotations.Test;
+
+/**
+ * @author Herman Kulik
+ */
+public class WaiterTest {
+    private static final int THRESHOLD = 30;
+    private static int delay;
+
+    @Factory(dataProvider = "dataMethod")
```
[the remark this answers is not in the document]
Comment (Hermix-art) [reply src/test/java/com/epam/prejap/tetris/game/WaiterTest.java:17]: Groups are added now

File: testing.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+
+<suite name="tests" verbose="1" >
+
+    <test name="delays" >
```
[the remark this answers is not in the document]
Comment (Hermix-art) [reply testing.xml]: Groups are added now

File: src/test/java/com/epam/prejap/tetris/game/WaiterTest.java (hunk 1 above)
Comment (Hermix-art) [reply src/test/java/com/epam/prejap/tetris/game/WaiterTest.java]: Yep, delay was carried out, so removed it from DPs

File: pom.xml [hunk 5]
```diff
@@ -28,17 +35,61 @@
...
+                            <mainClass>${main.class}</mainClass>
                         </manifest>
                     </archive>
                 </configuration>
             </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.0.0-M5</version>
```
Comment (marcin-ogorzalek-EPAM) [inline pom.xml:52]: Why not in `<properie>`?

File: pom.xml [hunk 6]
```diff
@@ -28,17 +35,61 @@
...
+                <configuration>
+                    <excludedGroups>jar</excludedGroups>
+                </configuration>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <version>3.0.0-M5</version>
```
Comment (marcin-ogorzalek-EPAM) [inline pom.xml:61]: Why not in `<properties>`?

**Comment by marcin-ogorzalek-EPAM** [comment]
@Hermix-art there are plenty of classes added here. Some even under your nicknames like `@author Grzegorz Konopka` :D  

Check the correctness of the Git history and try to fix it, so this MR will add only your code, not the one already present in the repository.

**Pull request closed without merging**